src/connection.js

```diff
@@ -18,28 +19,55 @@ var Session = require('./session');
  * @constructor
  */
 function Connection(id, options) {
-  Connection.super_.call(this, id);
+  /** @type {String} */
+  this._id = id;
 
+  /** @type {Object} */
   this._options = options || {};
 
-  /** @type {Transactions} */
-  this._transactions = new Transactions();
-
+  /** @type {Object} */
   this._sessions = {};
+
+  /** @type {WebsocketConnection} */
+  this._websocketConnection = new WebsocketConnection();
+  this._installWebsocketListeners();
 }
 
-util.inherits(Connection, WebsocketConnection);
+Helpers.extend(Connection.prototype, TEventEmitter, TTransactionGateway);
```

Comment:
This is a very fragile construction. You can not simply copy EventEmitter into Connection.prototype and expect that Connection will have emitter's API. EventEmitter is also a private state which can't be copied by simple extend. The approach that you can use is `Helpers.extend(Connection.prototype, EventEmitter.prototype, TTransactionGateway);` but also you need to call for EventEmitter's constructor in Connection's constructor.

```js
function Connection(id, options) {
  EventEmitter.call(this);
```
which is not a Trait. It's a partial multiple inheritance.